capstone-app/src/app/modules/history/wiki-media/wiki-media.component.spec.ts

```diff
@@ -53,12 +125,38 @@ describe('WikiMediaComponent', () => {
 
     it('displays location history', () => {
       component.getResults('Stillwater, Oklahoma');
-      expect(component.history).toEqual(testWikiResult.parse.text["*"]);
+      expect(component.history).toEqual(testWikiServiceResponse.history);
     });
 
     it('displays correct location title', () => {
       component.getResults('Stillwater, Oklahoma');
-      expect(component.title).toEqual("Stillwater, Oklahoma");
+      expect(component.title).toEqual(testWikiServiceResponse.title);
+    });
+
+    it('returns map of langlinks that can be read', () => {
+      component.getResults('Stillwater, Oklahoma');
+      expect(component.langlinks).toEqual(testWikiServiceResponse.langlinks);
+    });
+
+    it('language change function is called on button selection', () => {
+      spyOn(component, 'changeLanguage');
+      component.onChangeLanguage('de');
+      expect(component.changeLanguage).toHaveBeenCalledWith('Stillwater, (Oklahoma)', 'de', 'Geschichte');
+    });
+
+    it('history is changed correctly on changeLanguage call', () => {
```

Comment:
Optional: This is controversial, but since these three tests have identical setup, you can have just one test with three expects:

```
    it('history is changed correctly on changeLanguage call', () => {
 

 
Optional: This is controversial, but since these three tests have identical setup, you can have just one test with three expects:

```
    it('component is changed correctly on changeLanguage call', () => {
      component.changeLanguage('Stillwater, (Oklahoma)', 'de', 'Geschichte');
      expect(component.history).toEqual(testWikiGermanServiceResponse.history);
      expect(component.title).toEqual(testWikiGermanServiceResponse.title);
      expect(component.langlinks).toEqual(testWikiGermanServiceResponse.langlinks);
    });
```

Reply:
Oh, I see. I would prefer to leave them separate so if they fail it's more clear which one is failing. Is that okay?